agents/cache/utils.go

```diff
@@ -0,0 +1,75 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// MetricsFromStats builds Prometheus metrics from cache.Stats.
+func MetricsFromStats(stats Stats, agentID string, cacheType string) []prometheus.Metric {
+	metrics := []prometheus.Metric{
+		prometheus.MustNewConstMetric(mCurrentDesc, prometheus.GaugeValue, float64(stats.Current), agentID, cacheType),
+		prometheus.MustNewConstMetric(mUpdatedNDesc, prometheus.CounterValue, float64(stats.UpdatedN), agentID, cacheType),
+		prometheus.MustNewConstMetric(mAddedNDesc, prometheus.CounterValue, float64(stats.AddedN), agentID, cacheType),
+		prometheus.MustNewConstMetric(mRemovedNDesc, prometheus.CounterValue, float64(stats.RemovedN), agentID, cacheType),
+		prometheus.MustNewConstMetric(mOldestDesc, prometheus.CounterValue, float64(stats.Oldest.Unix()), agentID, cacheType),
+		prometheus.MustNewConstMetric(mNewestDesc, prometheus.CounterValue, float64(stats.Newest.Unix()), agentID, cacheType),
```

Comment:
Let's just move it inside `Collect` function, like we do it here: https://github.com/percona/pmm-managed/blob/1f20300018046bb68eb18878dab846a24e477ecc/services/agents/registry.go#L410